pkg/cmd/server/start/master_args.go

```diff
@@ -416,6 +416,17 @@ func (args MasterArgs) GetServerCertHostnames() (util.StringSet, error) {
 	}
 
 	allHostnames := util.NewStringSet("localhost", "127.0.0.1", "openshift.default.local", "kubernetes.default.local", "kubernetes-ro.default.local", masterAddr.Host, masterPublicAddr.Host, assetPublicAddr.Host)
+
+	// add the IPs that might be used based on the ListenAddr.
+	if args.ListenArg.ListenAddr.Host == "0.0.0.0" {
```

Comment:
do `ip := net.ParseIP(ipString)` and check `ip != nil && ip.IsUnspecified()` rather than "0.0.0.0"